addons/product/models/product.py

```diff
@@ -645,12 +645,11 @@ def price_compute(self, price_type, uom=None, currency=None, company=None, date=
         company = company or self.env.company
         date = date or fields.Date.context_today(self)
 
-        self = self.with_company(company)
         if price_type == 'standard_price':
             # standard_price field can only be seen by users in base.group_user
             # Thus, in order to compute the sale price from the cost for users not in this group
             # We fetch the standard price as the superuser
-            self = self.sudo()
+            self = self.with_company(company).sudo()
```

Comment:
In fact, this is not the right solution ...
It fixes your reproduceable version of the problem, but without fixing the root cause.

The problem comes from the fact that the rpc call to `/sale/get_combination_info` is not done with the right context, as the current company is not specified in the rpc call, which means `self.env.company = self.env.user.company_id = main company` & `self.env.companies = self.env.user.company_ids = (main company, second company)`, until `with_company` is called, after which `self.env.companies = self.env.company = main company` and removes the access to the second company.

If the right company was given as rpc context, `self.env.company = self.env.companies = second company` during the whole request, and everything should work fine (furthermore, currently, if a pricelist rule is based on the cost, the cost used as base would the cost specified in the user main company, not the current company, which would be fixed as well if the correct context is given to the rpc call).